dbt_platform_helper/commands/environment.py

```diff
@@ -594,50 +562,56 @@ def add_maintenance_page(
     env: str,
     services: List[Service],
     allowed_ips: tuple,
-    ip_filter: bool,
     template: str = "default",
 ):
     lb_client = session.client("elbv2")
     maintenance_page_content = get_maintenance_page_template(template)
+    bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
+
+    counter = 1
 
     for svc in services:
-        target_group_arn = find_target_group(app, env, svc.name)
+        target_group_arn = find_target_group(app, env, svc.name, session)
 
         # not all of an application's services are guaranteed to have been deployed to an environment
         if not target_group_arn:
             continue
 
-        if not ip_filter:
-            user_ip = get_public_ip()
-            allowed_ips = list(allowed_ips) + [user_ip]
-
+        allowed_ips = list(allowed_ips)
+        for ip_index, ip in enumerate(allowed_ips):
+            forwarded_rule_priority = counter * (ip_index + 100)
             create_header_rule(
                 lb_client,
                 listener_arn,
                 target_group_arn,
                 "X-Forwarded-For",
-                allowed_ips,
+                [ip],
                 "AllowedIps",
-            )
-        else:
-            bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
-            create_header_rule(
-                lb_client,
-                listener_arn,
-                target_group_arn,
-                "Bypass-Key",
-                [bypass_value],
-                "BypassIpFilter",
+                forwarded_rule_priority,
             )
 
-            click.secho(
-                f"\nUse a browser plugin to add `Bypass-Key` header with value {bypass_value} to your requests. For more detail, visit https://platform.readme.trade.gov.uk/ ",
-                fg="green",
-            )
+        bypass_rule_priority = counter
+        create_header_rule(
+            lb_client,
+            listener_arn,
+            target_group_arn,
+            "Bypass-Key",
+            [bypass_value],
+            "BypassIpFilter",
+            bypass_rule_priority,
+        )
+
+        counter += 1
+
+        click.secho(
+            f"\nUse a browser plugin to add `Bypass-Key` header with value {bypass_value} to your requests. For more detail, visit https://platform.readme.trade.gov.uk/activities/holding-and-maintenance-pages/",
+            fg="green",
+        )
 
+    fixed_rule_priority = 20000
```

Comment:
Just in case we had a squillion services...
```suggestion
    fixed_rule_priority = ((service_number + 5) * max_allowed_ips)
```

For the three services example, that would yield `800`, which should be obviously enough no related to one of the services.